File: test/core/mpi/test_mpi.cpp
// SPDX-License-Identifier: MIT
// SPDX-FileCopyrightText: 2023 NeoFOAM authors

#include <catch2/catch_test_macros.hpp>
#include "NeoFOAM/core/mpi/operations.hpp"


TEST_CASE("Send and receive data to and from a specific rank", "[mpi]")
{
    int rank;
    MPI_Comm_rank(MPI_COMM_WORLD, &rank);

    SECTION("subDict")
    {
        NeoFOAM::scalar value = 1.0;
        NeoFOAM::mpi::reduceAllScalar(&value, NeoFOAM::mpi::ReduceOp::Max, MPI_COMM_WORLD);
    }
}

File: test/telemetry/test_mpi.py
# SPDX-License-Identifier: GPL-3.0-or-later
# SPDX-FileCopyrightText: 2026 NeoFOAM authors


from neofoam.telemetry.mpi import current_mpi_info  # noqa: E402


def test_current_mpi_info_serial() -> None:
    mpi = current_mpi_info()
    assert mpi.rank == 0
    assert mpi.size == 1
    assert mpi.par_run is False
